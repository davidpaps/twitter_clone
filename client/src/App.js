import React, { Fragment, useState } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import "./App.css";
import SignIn from "./components/signIn/signIn";
import SignUp from "./components/signUp/signUp";
import Twitter from "./components/twitter/twitter";
import Navigation from "./components/navigation/navigation";
import Error from "./components/error/error";
import Home from "./components/home/home";
import SignOut from "./components/signOut/signOut";
import PrivateRoute from "./components/privateRoute";

const App = () => {
  const [auth, setAuth] = useState(false);

  return (
    <Fragment>
      <Router>
        <Navigation />
        <div className="container">
          <ul>
            <Switch>
              <Route exact path="/" component={Home} />
              <Route exact path="/sign_in" component={SignIn} />
              <Route exact path="/sign_up" component={SignUp} />
              <Route exact path="/sign_out" component={SignOut} />
              <PrivateRoute exact path="/home" component={Twitter} />
              <Route component={Error} />
            </Switch>
          </ul>
        </div>
      </Router>
    </Fragment>
  );
};

export default App;
