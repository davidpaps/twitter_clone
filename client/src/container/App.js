import React, { Fragment, useState } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import "./App.css";
import SignIn from "../components/home/signIn/signIn";
import SignUp from "../components/home/signUp/signUp";
import Twitter from "../components/twitter/twitter";
import Navigation from "../components/navigation/navigation";
import Error from "../components/error/error";
import Home from "../components/home/home";
import SignOut from "../components/navigation/signOut/signOut";
import PrivateRoute from "../components/privateRoute/privateRoute";

const App = () => {
  const [auth, setAuth] = useState(false);
  const [username, setUsername] = useState("");

  return (
    <Fragment>
      <Router>
        <Navigation auth={auth} username={username} />
        <div className="container">
          <ul>
            <Switch>
              <Route exact path="/">
                <Home />
              </Route>
              <Route
                exact
                path="/sign_in"
                render={(props) => (
                  <SignIn {...props} auth={setAuth} username={setUsername} />
                )}
              />
              <Route
                exact
                path="/sign_up"
                render={(props) => (
                  <SignUp {...props} auth={setAuth} username={setUsername} />
                )}
              />
              <Route exact path="/sign_out">
                <SignOut />
              </Route>
              <PrivateRoute
                exact
                path="/home"
                component={Twitter}
                username={username}
                auth={auth}
                setAuth={setAuth}
              />
              <Route>
                <Error />
              </Route>
            </Switch>
          </ul>
        </div>
      </Router>
    </Fragment>
  );
};

export default App;
