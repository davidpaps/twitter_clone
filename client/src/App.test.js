import React from "react";
import { shallow } from "enzyme";
import App from "./App";
import SignIn from "./components/signIn/signIn";

describe("App", () => {
  let wrapper;

  beforeEach(() => (wrapper = shallow(<App />)));

  it("should render a <div />", () => {
    expect(wrapper.find("div").length).toEqual(1);
  });

  it("should render the SignIn component", () => {
    expect(wrapper.containsMatchingElement(<SignIn />)).toEqual(true);
  });
});
