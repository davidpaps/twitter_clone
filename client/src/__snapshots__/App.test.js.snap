// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should rendercorrectly 1`] = `
Array [
  <div>
    <nav
      className="navbar navbar-expand-lg navbar-light bg-light"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        <img
          alt=""
          className="d-inline-block align-top"
          height="30"
          loading="lazy"
          src="https://vignette.wikia.nocookie.net/logopedia/images/9/99/Twitter_2012.svg/revision/latest/scale-to-width-down/340?cb=20180321032720"
          width="30"
        />
        Twitter
      </a>
      <div
        className="navbar-collapse collapse"
      />
      <div>
        <ul
          className="nav navbar-nav navbar-center"
        >
          <li>
            
          </li>
        </ul>
      </div>
      <div>
        <ul
          className="nav navbar-nav navbar-right"
        >
          <li
            className="nav-item active"
          >
            <a
              className="nav-link"
              href="/sign_up"
            >
              Sign In 
              <span
                className="sr-only"
              >
                (current)
              </span>
              <img
                alt=""
                className="d-inline-block "
                height="20"
                loading="lazy"
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTg9NgXjYkbVD1Wb4LzGSv9nYR6bC3Z_B4qWw&usqp=CAU"
                width="20"
              />
            </a>
          </li>
          <li
            className="nav-item active"
          >
            <a
              className="nav-link"
              href="/sign_in"
            >
              Sign Up 
              <span
                className="sr-only"
              >
                (current)
              </span>
              <img
                alt=""
                className="d-inline-block"
                height="20"
                loading="lazy"
                src="https://cdn3.iconfinder.com/data/icons/user-interface-2-9/34/169-512.png"
                width="20"
              />
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>,
  <div
    className="container"
  >
    <ul>
      <h1
        className="text-center mt-5"
      >
        Welcome to Twitter!
      </h1>
      <a
        href="/sign_up"
        onClick={[Function]}
      >
        <button
          className="btn btn-outline-success btn-sm mr-xl-5 w-100 h-100"
        >
          New User? Create an Account!
        </button>
      </a>
      <br />
      <br />
      <a
        href="/sign_in"
        onClick={[Function]}
      >
        <button
          className="btn btn-outline-info btn-sm mr-xl-5 w-100 h-100"
        >
          Already Have an Account? Click Here to Sign In!
        </button>
      </a>
    </ul>
  </div>,
]
`;
